src/app/Caches/DataDistribution.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import {
+    Card,
+    CardBody,
+    CardTitle,
+    Level,
+    LevelItem,
+    Select,
+    SelectOption,
+    SelectVariant,
+    Spinner,
+} from '@patternfly/react-core';
+import { Chart, ChartBar, ChartGroup, ChartVoronoiContainer } from '@patternfly/react-charts';
+import { TableErrorState } from '@app/Common/TableErrorState';
+import { useTranslation } from 'react-i18next';
+import { CustomCardTitle } from "@app/Common/CustomCardTitle";
+import { useDataDistribution } from '@app/services/dataDistributionHook';
+
+const DataDistribution = (props: {
+    cacheName: string;
```

Comment:
check if we need this parameter or we can use the context provider of cache detail and a hook to retrieve all

Reply:
I guess we need the parameter. It was passed from routes to parent components of this component.